pkg/utils/workloadidentity/workloadidentity_suite_test.go

```diff
@@ -2,7 +2,7 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-package workloadidentity_test
+package workloadidentity
```

Comment:
What is the reason to not use `workloadidentity_test`?

Reply:
It seems weird that this function is in a `_test` package but then tests actually use internals of that package. See https://github.com/gardener/gardener/blob/8f04d573e1c9f83d8723d0752df8d723a1f81606/pkg/utils/workloadidentity/jwt_test.go.

Reply:
Is there a requirement that all `*_test.go` files must be of the same package?

Reply:
No, I think not. An interesting approach that I found in the go std lib https://github.com/golang/go/blob/master/src/bufio/export_test.go. A file that is part of the package itself is used in order to export functions for testing. Then this file has a `_test` suffix so that it is not included in the output binary.

I can apply this "export" approach or revert the change that I made with regards to the package naming. WDYT?

Reply:
Applying the test export pattern seems nice, if it is not much work, it will be nice contribution. Otherwise I am fine with just reverting to `_test` package name.

Reply:
Done with https://github.com/gardener/gardener/pull/10383/commits/14dd01874511562769c7676bcc440e14da4e081c